stubs/Werkzeug/werkzeug/datastructures.pyi

```diff
@@ -191,7 +191,7 @@ class Headers(object):
     def extend(self, iterable): ...
     def __delitem__(self, key: Any) -> None: ...
     def remove(self, key): ...
-    def pop(self, **kwargs): ...
+    def pop(self, key: str, default: Optional[str] = ...): ...
```

Comment:
From the implementation, looks like the key can also be an int or None.

Also, default can really be of any type. For a precise return type you'll probably need two overloads: one that doesn't take default as an argument and returns `str`, and one that does and returns the Union of the default's type and the normal return type. `get()` already had some similar overloads, though it's more complicated because of other parameters.

Reply:
```python
@overload
def pop(self, key: str, default: Optional[str] = ...) -> str: ...
@overload
def pop(self, key: Optional[int], default: Optional[Tuple[str, str]] = ...) -> Tuple[str, str]: ...
```

What if I do something like this ? This is similar to the types present in the [original code](https://github.com/pallets/werkzeug/blob/bb8d422329750f36575ba19597240ff798fbb268/src/werkzeug/datastructures.pyi#L243). Also, will need to handle this for the `ImmutableHeadersMixin` class's `pop` method as well.

Reply:
Interesting, didn't realize the return type changed if the key is an int. Yes, let's just go with the same types the library itself provides.

We'll eventually drop the Werkzeug stubs in typeshed once there's a release with its own types (#5337).